__tests__/app.test.js

```diff
@@ -133,3 +133,122 @@ describe("/api/articles/:article_id/comments", () => {
+          author: expect(newComment.username),
+          body: expect(newComment.body),
+          votes: expect(0),
+          article_id: expect(1),
+        })
+      );
+  });
+
+  test("when given a article_id that doesn't exist return a 'not-found' message", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "best comment ever.",
+    };
+    return request(app)
+      .post("/api/articles/299/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("not found");
+      });
+  });
+
+  test("should return a sql error if the username doesn't exist in the users table ", () => {
+    const newComment = {
+      username: "vaish",
+      body: "best comment ever.",
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("bad request");
+      });
+  });
+});
+
+describe("PATCH /api/article/:article_id", () => {
+  test("When given an object with the keys inc_votes does it increase/decrease the votes and return a status 200", () => {
+    const voteObj = { inc_votes: 10 };
+    const updatedArticle = {
+      title: "Moustache",
+      topic: "mitch",
+      author: "butter_bridge",
+      body: "Have you seen the size of that thing?",
+      created_at: 1602419040000,
+      votes: 10,
+    };
+    return request(app)
+      .patch("/api/articles/12")
+      .send(voteObj)
+      .expect(200)
+      .then(({ body: { article } }) => {
+        expect.objectContaining({
+          title: expect("Moustache"),
+          topic: expect("mitch"),
+          author: expect("butter_bridge"),
+          body: expect("Have you seen the size of that thing?"),
+          votes: expect(10),
+        });
+      });
+  });
+
+  test("when given an article that doesn't exist then it returns a 404", () => {
+    const voteObj = { inc_votes: 10 };
+    const updatedArticle = {
+      title: "Moustache",
+      topic: "mitch",
+      author: "butter_bridge",
+      body: "Have you seen the size of that thing?",
+      created_at: 1602419040000,
+      votes: 10,
+    };
+    return request(app)
+      .patch("/api/articles/100")
+      .send(voteObj)
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("not found");
+      });
+  });
+});
+
+describe("09 GET api/users", () => {
+  test("when given the api/users endpoint returns an array of object and status 200", () => {
+    return request(app)
+      .get("/api/users")
+      .expect(200)
+      .then(({ body: { users } }) => {
+        users.forEach((user) => {
+          expect.objectContaining({
+            username: expect.any(String),
+            name: expect.any(String),
+            avatar_url: expect.any(String),
+          });
+        });
+      });
+  });
+
+  test("when given a api/user end point it should return a status 404", () => {
```

Comment:
We do not need to test for this as it would be a "different" path and handled accordingly 🙂